src/object_storage/objects.rs

```diff
@@ -234,6 +248,49 @@ impl IntoFallibleIterator for ObjectQuery {
     }
 }
 
+impl<R: Read + Send + 'static> NewObject<R> {
+    /// Start creating an object.
+    pub(crate) fn new(session: Rc<Session>, c_name: String, name: String, body: R) -> NewObject<R> {
+        NewObject {
+            session,
+            name,
+            c_name,
+            body,
+            delete_after: None,
+            delete_at: None,
```

Comment:
As I comment in `protocol.rs`, you could just have `headers` here and populate it on fly.

Reply:
So I have created a structure `ObjectHeaders` to store all optional headers related to an object. And then the headers are given as param to `api::create_object`